### PR TITLE
fix(api): event name types

### PR DESCRIPTION
**Pull request by 2nthony**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information

https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html

May a better dev experience with this type declaration, we can get code completion instead of `import { TauriEvent } from 'xxx'/event`, and the `TauriEvent["xxx"]` still works.

Tested in source code.

image

### CODE REVIEW

File: tooling/api/src/event.ts
Comment (amrbashir) [inline tooling/api/src/event.ts]: We also need to allow `string` type to be passed, `TauriEvent` is just for auto-completion purposes and not enforce a variant of it. I think we should be using this little trick from type-fest's [`LiteralUnion`](https://github.com/sindresorhus/type-fest/blob/main/source/literal-union.d.ts) instead

```suggestion
export type EventName = `${TauriEvent}` | (string & Record<never, never>)
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by amrbashir — APPROVED** [review]

**Comment by 2nthony** [comment]
Will this merge or not?🤔 @amrbashir

**Comment by amrbashir** [comment]
Sorry for the delay but the lead developer is on vacation atm and it is waiting for his approval.

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**
